test/ec_driver.cpp

```diff
@@ -78,23 +79,38 @@ void create_coding_files(
     std::vector<quadiron::Properties> c_props(fec->n_outputs);
 
     for (unsigned i = 0; i < fec->n_data; i++) {
-        snprintf(filename, sizeof(filename), "%s.d%d", prefix, i);
+        snprintf(filename, sizeof(filename), "%s.d%0*d", prefix, zpad, i);
```

Comment:
Not a big fan of `snprintf` here, it's more C than C++ and it can truncate the output (and we don't check the return value).

This formatting could be done using a `stringstream`. Example:

```C++
#include <cstdio>
#include <iomanip>
#include <iostream>
#include <sstream>


static inline
std::string get_filename(const std::string& pfx, int part, int zpad)
{
    std::ostringstream oss;
    oss << pfx << ".d" << std::setfill('0') << std::setw(zpad) << part;
    return oss.str();
}

int main () {
    int zpad = 10;
    int part = 42;
    char filename[1024];
    const char* prefix = "foo";

    snprintf(filename, sizeof(filename), "%s.d%0*d", prefix, zpad, part);
    std::cout << filename << '\n';

    std::cout << get_filename(prefix, part, zpad) << '\n';
    return 0;
}
```

Same for the other occurrences.

Reply:
OK thanks for the comments, I will address this.